sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDB.scala

```diff
@@ -72,7 +72,8 @@ class RocksDB(
     localRootDir: File = Utils.createTempDir(),
     hadoopConf: Configuration = new Configuration,
     loggingId: String = "",
-    useColumnFamilies: Boolean = false) extends Logging {
+    useColumnFamilies: Boolean = false,
+    ifEnableCheckpointId: Boolean = false) extends Logging {
```

Comment:
More consistent to call it `enableStateStoreCheckpointIds`.

I also think that the term "checkpoint ID" is very confusing. The term makes it feel like it's an ID for an _entire_ checkpoint, when really it's an ID for a particular state store that has been checkpointed.

I know it's a tedious modification to make. I would be happy to alleviate some of this work by creating a branch with that change and putting up a PR that you can merge back in this branch.

Reply:
Do you have a suggestion for a better name? I can definitely change it.

Reply:
State store checkpoint ID?